spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_dependant_child_attributes_spec.rb

```diff
@@ -39,10 +40,134 @@ module Requestors # rubocop:disable Metrics/ModuleLength
+            let!(:older_child) { create :dependant, :child16_to18, legal_aid_application: legal_aid_application, number: 2, has_income: true, assets_value: 0 }
 
             context 'variable attributes' do
+              context 'attribute CLI_RES_PER_INPUT_T_12WP3_1A - Person residing: name' do
+                let(:blocks) { XmlExtractor.call(xml, :client_residing_person, 'CLI_RES_PER_INPUT_T_12WP3_1A') }
+                let(:sorted_blocks) { blocks.sort { |a, b| a.css('ResponseValue').text <=> b.css('ResponseValue').text } }
+
+                it 'generates one block per dependant regardless of whether they are adults or children' do
+                  expect(blocks.size).to eq dependants.size
+                end
+
+                it 'includes block with name' do
+                  expected_names = dependants.map(&:name).sort
+                  sorted_blocks.each_with_index do |block, i|
+                    expect(block).to have_text_response expected_names[i]
+                  end
+                end
+              end
+
+              context 'attribute CLI_RES_PER_INPUT_B_12WP3_30A - Person residing: Employed?' do
+                let(:blocks) { XmlExtractor.call(xml, :client_residing_person, 'CLI_RES_PER_INPUT_B_12WP3_30A') }
+
+                it 'only generates a block for the adult_dependants' do
+                  expect(blocks.size).to eq dependants.adult_relative.size
+                end
+
+                it 'generates false for every block' do
+                  blocks.each do |block|
+                    expect(block).to have_boolean_response false
+                  end
+                end
+              end
+
+              context 'attribute CLI_RES_PER_INPUT_B_12WP3_31A - Dependant: receive their own income?' do
+                let(:blocks) { XmlExtractor.call(xml, :client_residing_person, 'CLI_RES_PER_INPUT_B_12WP3_31A') }
+
+                it 'generates a block only for adult dependants' do
+                  expect(blocks.size).to eq dependants.adult_relative.size
+                end
+
+                it 'generates false for every block' do
+                  blocks.each do |block|
+                    expect(block).to have_boolean_response false
+                  end
+                end
+              end
+
+              context 'attribute CLI_RES_PER_INPUT_T_12WP3_17A - Person residing: relationship to client' do
+                before do
+                  legal_aid_application.dependants.map(&:destroy!)
+                  create :dependant, legal_aid_application: legal_aid_application, date_of_birth: dob, relationship: relationship
+                  legal_aid_application.reload
+                end
+
+                context 'adult_relative' do
+                  let(:relationship) { 'adult_relative' }
+                  let(:dob) { Time.current - 22.years }
```

Comment:
I think for DOB it would be better to use Date.today instead of Time.current e.g `Date.today - 22.years`

So we are checking  `9 July 2001` instead of` 2001-07-09 15:50:05 +0100`